### ISSUE TITLE
Show deprecation warning in completion list

### ISSUE THREAD

**Issue by TonalidadeHidrica** [neoclide/coc.nvim#3480]
**Is your feature request related to a problem? Please describe.**
When I choose an item from a completion list, not until a warning is shown that I realize that the selected item is actually deprecated.

**Describe the solution you'd like**
I want a deprecation mark like ⚠️ shown before the item, or a strikethrough over the item name.

**Describe alternatives you've considered**
We may alternatively (or also) show that it is deprecated in the docs window shown when we pre-select the completion.

**Additional context**
Specifically I want this feature for [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer).  I first [asked `rust-analyzer` for this feature](https://github.com/rust-analyzer/rust-analyzer/issues/10844) to be informed that the LSP already provides such diagnostic and is available in VSCode.   (A screenshot is included in that issue.)

**Comment by oblitum** [comment]
coc.nvim already does this, if it comes as warning, you can check that with typescript + coc-tsserver for example.

What it doesn't do is this:
- https://github.com/neoclide/coc.nvim/pull/3188#issuecomment-974490202

**Comment by TonalidadeHidrica** [comment]
I realized that I wrote a little bit confusing description, sorry for that.  So let me summarize; let me know if I said something wrong.

There are at least three ways for Language Servers to tell that an item is deprecated:
- Through diagnostics.  LS points out through diagnostics that some part of a code is using deprecated function etc..
  - This can also be viewed in `:CocList diagnostics`.
  - You pointed out that it is already available in typescript + coc-tsserver, right?  Yes, it indeed works in `coc-rust-analyzer`, too.  
- Through semantic tokens.  That is a concept I have never heard of, but perhaps that is an additional information attached to tokens which are sent from LS as a result of its tokenization, isn't it?
- Through completion tags.  I didn't know that either, but probably this *is* what I want, which probably comes with each entry in completion list given from LS.

And you said that the latter two is not entirely supported by coc.nvim.

Is there already an issue that specifically ask for this kind of features in *completion list*?  If there is, please close the issue; otherwise I'll leave it as a ticket of feature request.

**Comment by oblitum** [comment]
You can keep the ticket, there isn't an specific issue that covers either of the two latter ones.

**Comment by TonalidadeHidrica** [comment]
@oblitum Thanks.  I additionally filed another issue on semantic tokens in #3481.

**Comment by oblitum** [comment]
@TonalidadeHidrica okay. Notice an strikethrough over some pum items is in fact not possible due to vim/nvim limitation, pum doesn't support syntax highlighting. So the only way out (with pum) is by means of a glyph mark.

**Comment by oblitum** [comment]
To have that with completion you have to ditch vim/nvim native completion system and reimplement it through vim-popups/nvim-floating-windows, like [nvim-cmp has done](https://github.com/hrsh7th/nvim-cmp).

**Comment by chemzqm** [comment]
Not too hard to implement completion pum by using float window or popup, but lots of details and break changes of how completion works.

Vote for the feature if you really like it.
